plugins/auth-backend-module-oidc-proxy-provider/config.d.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  auth?: {
+    providers?: {
+      /**
+       * Configuration for the oidc identity aware proxy auth provider.
+       */
+      oidcProxy?: {
+        [authEnv: string]: {
+          /**
+           * the iss claim value to assert validating an id token.
+           */
+          issuer: string;
+
+          /**
+           * the aud claim value to assert when validating an id token.
+           */
+          audience: string;
+
+          /**
+           * the header to read the id token from.  Defaults to `'x-oidc-id-token'`.
+           */
+          idTokenHeader?: string;
+        };
```

Comment:
maybe add the sign in resolvers here too?

example: https://github.com/backstage/backstage/blob/3110750a88d71f49a3debee6ae3a0af5d37b87c1/plugins/auth-backend-module-aws-alb-provider/config.d.ts#L42

Reply:
Added in 84309f706907 oidc-proxy-provider: clarify sign in provider config